backends/bmv2/jsonconverter.cpp

```diff
@@ -1724,13 +1817,38 @@ Util::IJson* JsonConverter::convertControl(const IR::ControlBlock* block, cstrin
                     jmtr->emplace("result_target", result->to<Util::JsonObject>()->get("value"));
                     meters->append(jmtr);
                     continue;
-                } else if (eb->type->name == v1model.action_profile.name) {
+                } else if (eb->type->name == v1model.action_profile.name ||
+                           eb->type->name == v1model.action_selector.name) {
                     auto action_profile = new Util::JsonObject();
                     action_profile->emplace("name", name);
                     action_profile->emplace("id", nextId("action_profiles"));
-                    auto sz = eb->getParameterValue(v1model.action_profile.sizeParam.name);
-                    BUG_CHECK(sz->is<IR::Constant>(), "%1%: expected a constant", sz);
-                    action_profile->emplace("max_size", sz->to<IR::Constant>()->value);
+
```

Comment:
It looks like much of this code is duplicated for the case where a ConstructorCallExpression is used.
It would have been nice to share this code, but it is not high priority.